vm/src/stdlib/array.rs

```diff
@@ -194,14 +195,31 @@ macro_rules! def_array_enum {
                 }
             }
 
+            fn getitem_by_slice(&self, slice: PySliceRef, vm: &VirtualMachine) -> PyResult<PyObjectRef> {
+                match self {
+                    $(ArrayContentType::$n(v) => {
+                        let elements = v.get_slice_items(vm, &slice)?;
+                        let zelf = ArrayContentType::$n(elements);
+                        let obj = PyObject::new(
+                            PyArray {
+                                array: PyRwLock::new(zelf)
+                            },
+                            PyArray::class(vm),
+                            None
+                        );
```

Comment:
`zelf` is an alternative name of `self` when it is not allowed by rust languge restriction. In this code, it doesn't mean self but a sliced one. New variable name like `sliced` or something else would be better